system/classes/KO7/Controller/REST.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * Abstract Controller class for REST controller mapping.
+ * Supports GET, PUT, POST, and DELETE.
+ *
+ * @package        KO7\REST
+ *
+ * @copyright  (c) since 2016 Koseven Team
+ * @license        https://koseven.ga/LICENSE
+ */
+abstract class KO7_Controller_REST extends Controller {
+
+	/**
+	 * Default Output (only if allowed by configuration)
+	 * @var string
+	 */
+	public static $default_output = 'json';
+
+	/**
+	 * The output format to be used (JSON, XML etc.).
+	 * @var string
+	 */
+	public $output_format;
+
+    /**
+     * REST types
+     * @var array
+     */
+    protected static $_action_map = [
+        HTTP_Request::GET => 'index',
+        HTTP_Request::PUT => 'update',
+        HTTP_Request::POST => 'create',
+        HTTP_Request::DELETE => 'delete'
+    ];
+
+    /**
+     * Instance of body formatting class
+     * @var REST_Format
+     */
+    protected $_formatter;
+
+    /**
+     * Allows overwriting the content_type in the response header
+     * If this is null, content type will be evaluated from the output_format
+     * @var string
+     */
+	protected $_content_type;
+
+	/**
+	 * The request's parameters.
+	 * @var array
+	 */
+	protected $_params = [];
+
+	/**
+	 * KO7_Controller_REST constructor.
+	 *
+	 * We need to convert the "Response" object to a "Response_REST" object.
+     * This allows as passing arrays and objects as content body and not only strings
+	 *
+	 * @param Request  $request   Request that created the controller
+	 * @param Response $response  The request's response
+	 */
+	public function	__construct(Request $request, Response $response)
+	{
+		// Copy class properties
+		$config = [
+			'_status'   => $response->status(),
+			'_body'     => $response->body(),
+			'_cookies'  => $response->cookie(),
+			'_protocol' => $response->protocol()
+		];
+
+		// Copy all http header(s)
+		foreach ($response->headers()->getArrayCopy() as $header => $value)
+		{
+			$config[$header] = $value;
+		}
+
+		// Instance Response_REST and pass it to the constructor
+		parent::__construct($request, Response_REST::factory($config));
+	}
+
+	/**
+     * Automatically executed before the controller action.
+     * Evaluate Request (method, action, parameter, format)
+     *
+     * @throws HTTP_Exception
+	 */
+	public function before() : void
+	{
+		// Parent call
+		parent::before();
+
+        // Try fetching method from HTTP_X_HTTP_METHOD_OVERRIDE, otherwise use the one sent with the request
+        $this->request->method($_SERVER['HTTP_X_HTTP_METHOD_OVERRIDE'] ?? $this->request->method());
```

Comment:
**always** use `filter_var()` for global variables

Reply:
Added inline check instead